code/modules/clothing/head/misc_special.dm

```diff
@@ -152,7 +152,28 @@
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0)
 	light_range = 2 //luminosity when on
 	flags_cover = HEADCOVERSEYES
+	light_color = "#fff2bf"
+	worn_y_offset = 1
 
+/obj/item/clothing/head/hardhat/pumpkinhead/update_overlays()
```

Comment:
```suggestion
	worn_y_offset = 1


/obj/item/clothing/head/hardhat/pumpkinhead/set_light_on(new_value)
	. = ..()
	if(isnull(.))
		return
	update_icon(UPDATE_OVERLAYS)


/obj/item/clothing/head/hardhat/pumpkinhead/update_overlays()
```
So that if the light is toggled on and off the overlay is properly updated.